agent/provider/src/cli/keystore.rs

```diff
@@ -116,7 +111,33 @@ fn add(config: ProviderConfig, add: Add) -> anyhow::Result<()> {
 fn remove(config: ProviderConfig, remove: Remove) -> anyhow::Result<()> {
     let cert_dir = config.cert_dir_path()?;
     let mut keystore = CompositeKeystore::load(&cert_dir)?;
-    let RemoveResponse { removed } = keystore.remove(&remove.into())?;
+
+    let all_certs = keystore.list();
+    let mut ids = HashSet::new();
+    for remove_prefix in &remove.ids {
+        let full_ids = find_ids_by_prefix(&all_certs, remove_prefix);
+
+        if full_ids.is_empty() {
+            ids.insert(remove_prefix.clone()); //won't match anyway
+        } else if full_ids.len() == 1 {
+            ids.insert(full_ids[0].clone());
+        } else {
+            println_conditional(
+                &config,
+                &format!(
+                    "Prefix '{remove_prefix}' isn't unique, consider using full certificate id"
+                ),
```

Comment:
Should't we have a following logic:
If prefix is not unique then extend prefixes?